### PR TITLE
Add `ReadonlyArray.sort()` macro

### PR DESCRIPTION
**Pull request by daniel-melanson**

### CODE REVIEW

**Comment by daniel-melanson** [comment]
TypeScript:
```ts
let arr = [1, 2, 3, 4];

arr.sort();

arr.sort((a, b) => a > b);
```
Lua:
```lua
table.sort(arr)
table.sort(arr, function(a, b)
	return a > b
end)
```

**Comment by Dionysusnu** [comment]
Isn't Lua's callback type slightly different to JavaScript's default one? Might be confusing.

**Comment by daniel-melanson** [comment]
Yes. JavaScript uses a numeric system. The sorting function returns a number. If that number is less than `0`, then `a` is less than `b`. If the number is `0`, `a` is equal to `b`. If the number is greater than `0`, then `a` is greater than `b`. 

Lua uses a boolean system. Return true if `a` is less than `b`.

However, the roblox-ts type definitions states: _@param compareFunction — A function that defines the sort order. Returns true when the first element must come before the second. If omitted, the array is sorted according to the < operator._ On top of that, there are type definitions.

Although it is confusing, using the type definitions, linters will catch cases in which the callback does not return a boolean.

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 1]
```diff
@@ -435,6 +435,18 @@ const READONLY_ARRAY_METHODS: MacroList<PropertyCallMacro> = {
 	findIndex: runtimeLib("array_findIndex"),
 	indexOf: runtimeLib("array_indexOf"),
 	find: runtimeLib("array_find"),
+	sort: (state, node, expression) => {
+		const list = lua.list.make(expression);
+
+		if (node.arguments?.length > 0) {
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:441]:
```suggestion
		if (node.arguments.length > 0) {
```
I don't think the `?` is necessary here. `node.arguments` should always be defined?

**Comment by osyrisrblx** [comment]
Oops, looks like I accidentally deleted a `}` in my merge conflict resolve.

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 2]
```diff
@@ -435,6 +435,18 @@ const READONLY_ARRAY_METHODS: MacroList<PropertyCallMacro> = {
 	findIndex: runtimeLib("array_findIndex"),
 	indexOf: runtimeLib("array_indexOf"),
 	find: runtimeLib("array_find"),
+	sort: (state, node, expression) => {
+		const list = lua.list.make(expression);
```
[the remark this answers is not in the document]
Comment (Dionysusnu) [reply src/TSTransformer/macros/propertyCallMacros.ts:480]: a bit unrelated, but, why does LuaAST used mixed `arguments` and `args`?

File: src/TSTransformer/macros/propertyCallMacros.ts (hunk 1 above)
Comment (daniel-melanson) [reply src/TSTransformer/macros/propertyCallMacros.ts:482]: Sort can be called without arguments correct?

Comment (daniel-melanson) [reply src/TSTransformer/macros/propertyCallMacros.ts:482]: Arguments will be set to an empty array if none.

**Pull request merged**